src/main/java/com/squareup/javapoet/AnnotationSpec.java

```diff
@@ -221,7 +223,9 @@ public Builder addMember(String name, CodeBlock codeBlock) {
      * depending on the given {@code value} object. Falls back to {@code "$L"} literal format if
      * the class of the given {@code value} object is not supported.
      */
-    private Builder addValue(String memberName, Object value) {
+    public Builder addMemberForValue(String memberName, Object value) {
+      checkArgument(memberName != null, "memberName == null");
+      checkArgument(value != null, "only constant non-null values are permitted (%s)", memberName);
```

Comment:
This came up very recently. Our rule is: NPE when null is not expected even if it is an argument, and IAE for non-null expectation violations.